src/components/AttachmentModal.tsx

```diff
@@ -196,17 +217,19 @@ function AttachmentModal(props) {
      * Download the currently viewed attachment.
      */
     const downloadAttachment = useCallback(() => {
-        let sourceURL = source;
-        if (isAuthTokenRequired) {
-            sourceURL = addEncryptedAuthTokenToURL(sourceURL);
+        let sourceURL = sourceState;
+        if (isAuthTokenRequiredState && typeof sourceURL === 'string') {
+            sourceURL = addEncryptedAuthTokenToURL(sourceURL ?? '');
```

Comment:
I think it should be ok since we check that `sourceURL` is a `string`
```suggestion
            sourceURL = addEncryptedAuthTokenToURL(sourceURL);
```